core/dbt/cli/options.py

```diff
@@ -0,0 +1,44 @@
+import click
+
+
+# Implementation from: https://stackoverflow.com/a/48394004
+# Note OptionEatAll options must be specified with type=tuple (https://github.com/pallets/click/issues/2012)
+class OptionEatAll(click.Option):
```

Comment:
adding this to support backwards compatability for `--select`, `--model`, `--exclude`, and `--resource-type` options. click supports a `multiple=True` setting for options that expects multiple values be set through multiple instances of the option, ie: 
`--select a --select b --select c`, as opposed to our existing `--select a b c`

There is an option for setting `nargs=-1` in click but this is limited to a single option, that is specified last in the command. 

confirmed with @jtcohen6 that not supporting the `--select a b c` syntax would represent too large of a breaking change for a minor release.